### PR TITLE
chore(jenkins/controller): changing jdk21 ea version to the ea line

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3736
moving to the official EA line of jdk21

### CODE REVIEW

**Review by timja — DISMISSED** [review]

**Review by dduportal — COMMENTED** [review]
Should we add the version in the top-level `tools_versions-version` (in https://github.com/jenkins-infra/jenkins-infra/blob/3596b5c265426f1ad63e209ddde3a3540e0851d0/hieradata/common.yaml#L252-L256) to make https://github.com/jenkins-infra/jenkins-infra/pull/3051 easier (and avoid multiple repetitions of the `version` and `subdir` attributes?

Ref. https://github.com/jenkins-infra/helpdesk/issues/3736#issuecomment-1699067801

**Comment by smerle33** [comment]
> Should we add the version in the top-level `tools_versions-version` (in
> 
> https://github.com/jenkins-infra/jenkins-infra/blob/3596b5c265426f1ad63e209ddde3a3540e0851d0/hieradata/common.yaml#L252-L256
> ) to make #3051 easier (and avoid multiple repetitions of the `version` and `subdir` attributes?
> 
> Ref. [jenkins-infra/helpdesk#3736 (comment)](https://github.com/jenkins-infra/helpdesk/issues/3736#issuecomment-1699067801)

not yet, as the unzip folder still need to be specified (no -ea-beta) in the folder name ....

File: dist/profile/templates/jenkinscontroller/casc/tools.yaml.erb [hunk 1]
```diff
@@ -56,7 +56,7 @@ tool:
             }
             -%>
               # subdir is the top-level directory in the archive: depends on the how the JDK is packaged. Defaults to adoptium convention unless a customized version is specified
-              # Example of subdirs for adoptium 8: jdk8u345-b01, adoptium 11: jdk-11.0.16+8, adoptium 17: jdk-17.0.4+8, adoptium 19: jdk-19.0.1+10, adoptium 21: jdk-21+34 ...
+              # Example of subdirs for adoptium 8: jdk8u345-b01, adoptium 11: jdk-11.0.16+8, adoptium 17: jdk-17.0.4+8, adoptium 19: jdk-19.0.1+10, adoptium 21: jdk-21+35git st ...
```
Comment (dduportal) [inline dist/profile/templates/jenkinscontroller/casc/tools.yaml.erb]:
```suggestion
              # Example of subdirs for adoptium 8: jdk8u345-b01, adoptium 11: jdk-11.0.16+8, adoptium 17: jdk-17.0.4+8, adoptium 19: jdk-19.0.1+10, adoptium 21: jdk-21+35...
```

**Comment by dduportal** [comment]
> > Should we add the version in the top-level `tools_versions-version` (in
> > https://github.com/jenkins-infra/jenkins-infra/blob/3596b5c265426f1ad63e209ddde3a3540e0851d0/hieradata/common.yaml#L252-L256
> > 
> > ) to make #3051 easier (and avoid multiple repetitions of the `version` and `subdir` attributes?
> > Ref. [jenkins-infra/helpdesk#3736 (comment)](https://github.com/jenkins-infra/helpdesk/issues/3736#issuecomment-1699067801)
> 
> not yet, as the unzip folder still need to be specified (no -ea-beta) in the folder name ....

Unless I'm mistaken, with the new `ea` distribution of temurin 21, the subdir is now having a fixed named: `jdk-21+35` (which is the version). So there is no need to specify a subdir for each tool installation kind: we can factorize like it's done for all the other JDKs (which is the goal of using the `ea` distribution)

**Comment by smerle33** [comment]
tested with vagrant with success path to download and subdir folder

**Review by dduportal — APPROVED** [review]
Verified with `vagrant` on my side:

- Started the "test" controller with `bash ./scripts/vagrant-bootstrap.sh && vagrant up --provision jenkins::controller`
- Accessed the Jenkins controler through http://localhost:8080
- In "Manage Jenkins", I've set up the "Jenkins Node" (controller itself) to have 2 executors and with the labels `linux` and `arm64` (space separated) to simulate an agent
- Ran the following pipeline:

  ```
  node('linux && arm64') {
      javaHome = tool name: 'jdk21', type: 'jdk'
      withEnv(["JAVA_HOME=${javaHome}"]){
          sh '"$JAVA_HOME"/bin/java --version'
      }
  }
  ```
  with the following result:
  ```
  Started by user unknown or anonymous
  Loading library pipeline-library@master
  Library pipeline-library@master is cached. Copying from home.
  [Pipeline] Start of Pipeline
  [Pipeline] node
  Running on Jenkins in /var/jenkins_home/workspace/test
  [Pipeline] {
  [Pipeline] tool
  [Pipeline] withEnv
  [Pipeline] {
  [Pipeline] sh
  + /var/jenkins_home/tools/hudson.model.JDK/jdk21/jdk-21+35/bin/java --version
  openjdk 21-beta 2023-09-19
  OpenJDK Runtime Environment Temurin-21+35-202309042128 (build 21-beta+35-ea)
  OpenJDK 64-Bit Server VM Temurin-21+35-202309042128 (build 21-beta+35-ea, mixed mode, sharing)
  [Pipeline] }
  [Pipeline] // withEnv
  [Pipeline] }
  [Pipeline] // node
  [Pipeline] End of Pipeline
  Finished: SUCCESS
  ```



Looks good to me 👏 

@smerle33 I let you merge, deploy and watch the result once you are ready

**Pull request merged**